### PR TITLE
Symbolic constraint refactoring

### PR DESCRIPTION
**Pull request by andreistefanescu**
do not review yet

### CODE REVIEW

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by andreistefanescu** [comment]
Jenkins: test this please

**Comment by andreistefanescu** [comment]
Jenkins: test this please

**Comment by andreistefanescu** [comment]
@yilongli @dwightguth please review

**Comment by yilongli** [comment]
lgtm

File: java-backend/src/main/java/org/kframework/backend/java/kil/KItem.java [hunk 1]
```diff
@@ -317,7 +315,7 @@ public Term resolveFunctionAndAnywhere(KItem kItem, boolean copyOnShareSubstAndE
                             kItem.applyAnywhereRules(copyOnShareSubstAndEval, context);
                 if (result instanceof KItem && ((KItem) result).isEvaluable(context) && result.isGround()) {
                     // we do this check because this warning message can be very large and cause OOM
-                    if (options.warnings.includesExceptionType(ExceptionType.WARNING) && tool != Tool.KOMPILE) {
+                    if (options.warnings.includesExceptionType(ExceptionType.HIDDENWARNING) && tool != Tool.KOMPILE) {
```
Comment (dwightguth) [inline java-backend/src/main/java/org/kframework/backend/java/kil/KItem.java:318]: if you want this to be a hidden warning, you should also modify the call to registerWarning, for consistency.

**Comment by dwightguth** [comment]
This diff has broken the rule auditing feature. I need you to go through the code, find every place where you deleted a piece of that infrastructure, and make sure that it is being called in the equivalent location in the new code.

**Comment by andreistefanescu** [comment]
@dwightguth you have to be a little more specific, as I have no idea what rule auditing does...

**Comment by dwightguth** [comment]
Specifically, I am referring to this block of code that was removed from the falsify method for constraints:

```
if (RuleAuditing.isAuditBegun()) {
 System.err.println("Unification failure: " + equality.leftHandSide()
 + " does not unify with " + equality.rightHandSide());
}
```

**Comment by andreistefanescu** [comment]
Thanks.

**Comment by dwightguth** [comment]
lgtm. Merge if tests pass.

**Comment by andreistefanescu** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Change broke C semantics. Reverting. Please fix and resubmit PR.

**Pull request merged**